### ISSUE TITLE
Create WordPress ACME challenge directory fails

### ISSUE THREAD

**Issue by r4nd0m123** [carlalexander/debops-wordpress#164]
Hi,

When I try to use `wordpress__ssl: True` in inventory/host_vars/test.domain.com/vars.yml it fails when it comes to the `Create WordPress ACME challenge directory` task.

The host is Ubuntu 16.04.2 running on Vultr. 

```
TASK [wordpress : Create WordPress ACME challenge directory] ********************************************************************************************************************************************************************************
task path: /home/user/debops-wordpress/roles/wordpress/tasks/manage_ssl.yml:3
skipping: [www.domain.com] => {
    "changed": false, 
    "skip_reason": "Conditional result was False", 
    "skipped": true
}
Using module file /home/user/.local/lib/python2.7/site-packages/ansible/modules/files/file.py
<test.domain.com> ESTABLISH SSH CONNECTION FOR USER: None
<test.domain.com> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/user/.ansible/cp/24aa300219 test.domain.com '/bin/sh -c '"'"'echo ~ && sleep 0'"'"''
<test.domain.com> (0, '/var/local/user\n', 'OpenSSH_7.2p2 Ubuntu-4ubuntu2.2, OpenSSL 1.0.2g  1 Mar 2016\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 5788\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<test.domain.com> ESTABLISH SSH CONNECTION FOR USER: None
<test.domain.com> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/user/.ansible/cp/24aa300219 test.domain.com '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /var/local/user/.ansible/tmp/ansible-tmp-1497876203.35-20843310008228 `" && echo ansible-tmp-1497876203.35-20843310008228="` echo /var/local/user/.ansible/tmp/ansible-tmp-1497876203.35-20843310008228 `" ) && sleep 0'"'"''
<test.domain.com> (0, 'ansible-tmp-1497876203.35-20843310008228=/var/local/user/.ansible/tmp/ansible-tmp-1497876203.35-20843310008228\n', 'OpenSSH_7.2p2 Ubuntu-4ubuntu2.2, OpenSSL 1.0.2g  1 Mar 2016\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 5788\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<test.domain.com> PUT /tmp/tmpjLBNep TO /var/local/user/.ansible/tmp/ansible-tmp-1497876203.35-20843310008228/file.py
<test.domain.com> SSH: EXEC sftp -b - -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/user/.ansible/cp/24aa300219 '[test.domain.com]'
<test.domain.com> (0, 'sftp> put /tmp/tmpjLBNep /var/local/user/.ansible/tmp/ansible-tmp-1497876203.35-20843310008228/file.py\n', 'OpenSSH_7.2p2 Ubuntu-4ubuntu2.2, OpenSSL 1.0.2g  1 Mar 2016\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 5788\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug2: Remote version: 3\r\ndebug2: Server supports extension "posix-rename@openssh.com" revision 1\r\ndebug2: Server supports extension "statvfs@openssh.com" revision 2\r\ndebug2: Server supports extension "fstatvfs@openssh.com" revision 2\r\ndebug2: Server supports extension "hardlink@openssh.com" revision 1\r\ndebug2: Server supports extension "fsync@openssh.com" revision 1\r\ndebug3: Sent message fd 6 T:16 I:1\r\ndebug3: SSH_FXP_REALPATH . -> /var/local/user size 0\r\ndebug3: Looking up /tmp/tmpjLBNep\r\ndebug3: Sent message fd 6 T:17 I:2\r\ndebug3: Received stat reply T:101 I:2\r\ndebug1: Couldn\'t stat remote file: No such file or directory\r\ndebug3: Sent message SSH2_FXP_OPEN I:3 P:/var/local/user/.ansible/tmp/ansible-tmp-1497876203.35-20843310008228/file.py\r\ndebug3: Sent message SSH2_FXP_WRITE I:4 O:0 S:32768\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 4 32768 bytes at 0\r\ndebug3: Sent message SSH2_FXP_WRITE I:5 O:32768 S:28816\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 5 28816 bytes at 32768\r\ndebug3: Sent message SSH2_FXP_CLOSE I:4\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<test.domain.com> ESTABLISH SSH CONNECTION FOR USER: None
<test.domain.com> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/user/.ansible/cp/24aa300219 test.domain.com '/bin/sh -c '"'"'chmod u+x /var/local/user/.ansible/tmp/ansible-tmp-1497876203.35-20843310008228/ /var/local/user/.ansible/tmp/ansible-tmp-1497876203.35-20843310008228/file.py && sleep 0'"'"''
<test.domain.com> (0, '', 'OpenSSH_7.2p2 Ubuntu-4ubuntu2.2, OpenSSL 1.0.2g  1 Mar 2016\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 5788\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<test.domain.com> ESTABLISH SSH CONNECTION FOR USER: None
<test.domain.com> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/user/.ansible/cp/24aa300219 -tt test.domain.com '/bin/sh -c '"'"'sudo -H -S -n -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-uhkgiwebqwhwrpjeyvxssodnhcxwnuio; /usr/bin/python /var/local/user/.ansible/tmp/ansible-tmp-1497876203.35-20843310008228/file.py; rm -rf "/var/local/user/.ansible/tmp/ansible-tmp-1497876203.35-20843310008228/" > /dev/null 2>&1'"'"'"'"'"'"'"'"' && sleep 0'"'"''
<test.domain.com> (0, '\r\n{"group": "root", "uid": 0, "failed": true, "state": "directory", "gid": 0, "mode": "0755", "invocation": {"module_args": {"directory_mode": null, "force": false, "remote_src": null, "path": "/var/www/acme/.well-known/acme-challenge", "owner": "pki-acme", "follow": false, "group": "pki-acme", "unsafe_writes": null, "state": "directory", "content": null, "serole": null, "diff_peek": null, "setype": null, "selevel": null, "original_basename": null, "regexp": null, "validate": null, "src": null, "seuser": null, "recurse": false, "delimiter": null, "mode": "0750", "attributes": null, "backup": null}}, "owner": "root", "path": "/var/www/acme", "size": 4096, "msg": "chown failed: failed to look up user pki-acme"}\r\n', 'OpenSSH_7.2p2 Ubuntu-4ubuntu2.2, OpenSSL 1.0.2g  1 Mar 2016\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 5788\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\nShared connection to test.domain.com closed.\r\n')
fatal: [test.domain.com]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "gid": 0, 
    "group": "root", 
    "invocation": {
        "module_args": {
            "attributes": null, 
            "backup": null, 
            "content": null, 
            "delimiter": null, 
            "diff_peek": null, 
            "directory_mode": null, 
            "follow": false, 
            "force": false, 
            "group": "pki-acme", 
            "mode": "0750", 
            "original_basename": null, 
            "owner": "pki-acme", 
            "path": "/var/www/acme/.well-known/acme-challenge", 
            "recurse": false, 
            "regexp": null, 
            "remote_src": null, 
            "selevel": null, 
            "serole": null, 
            "setype": null, 
            "seuser": null, 
            "src": null, 
            "state": "directory", 
            "unsafe_writes": null, 
            "validate": null
        }
    }, 
    "mode": "0755", 
    "msg": "chown failed: failed to look up user pki-acme", 
    "owner": "root", 
    "path": "/var/www/acme", 
    "size": 4096, 
    "state": "directory", 
    "uid": 0
}
```
There is no such _pki-acme_ user on the host. Is there a task that is responsible for creating this?

Thanks

**Comment by r4nd0m123** [comment]
I see others have encountered this issue. #133 
Manually creating the pki-acme user and group has resolved this issue but I too encountered another error ejholdings encountered.

```
TASK [wordpress : Restart nginx to pickup new user group] ***********************************************************************************************************************************************************************************
skipping: [other.domain.com]
fatal: [test.domain.com]: FAILED! => {"changed": false, "failed": true, "msg": "Unable to start service nginx: Job for nginx.service failed because the control process exited with error code. See \"systemctl status nginx.service\" and \"journalctl -xe\" for details.\n"}
```

```
● nginx.service - A high performance web server and a reverse proxy server
   Loaded: loaded (/lib/systemd/system/nginx.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2017-06-21 11:33:17 UTC; 35s ago
  Process: 26837 ExecStartPre=/usr/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=1/FAILURE)

Jun 21 11:33:17 test.domain.com systemd[1]: Starting A high performance web server and a reverse proxy server...
Jun 21 11:33:17 test.domain.com nginx[26837]: nginx: [emerg] ipv6only is not supported on addr "0.0.0.0:8080", ignored in /etc/nginx/sites-enabled/test.domain.com.conf:10
Jun 21 11:33:17 test.domain.com nginx[26837]: nginx: [emerg] BIO_new_file("/etc/pki/realms/test.domain.com/default.crt") failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/etc/pki/realms/test
Jun 21 11:33:17 test.domain.com nginx[26837]: nginx: configuration file /etc/nginx/nginx.conf test failed
Jun 21 11:33:17 test.domain.com systemd[1]: nginx.service: Control process exited, code=exited status=1
Jun 21 11:33:17 test.domain.com systemd[1]: Failed to start A high performance web server and a reverse proxy server.
Jun 21 11:33:17 test.domain.com systemd[1]: nginx.service: Unit entered failed state.
Jun 21 11:33:17 test.domain.com systemd[1]: nginx.service: Failed with result 'exit-code'.
```

**Comment by carlalexander** [comment]
Did you run the other two commands first? I think they create the `pki-acme` user. I haven't tested on Vultr so there could be a problem there.

**Comment by r4nd0m123** [comment]
I did run both commands first and they finished without any failures.

$debops bootstrap -u root
```
PLAY RECAP **********************************************************************************************************************************************************************************************************************************
other.domain.com   : ok=14   changed=1    unreachable=0    failed=0
test.domain.com     : ok=14   changed=1    unreachable=0    failed=0

TASK: debops.bootstrap/raw : Update APT repositories, install mandatory packages -- 19.22s
TASK: debops.bootstrap : Make sure essential software is installed ----- 18.53s
TASK: debops.bootstrap : Create required system groups ------------------ 8.79s
TASK: debops.bootstrap : Configure system group with passwordless access in sudo --- 7.55s
TASK: debops.bootstrap : Install ssh public key on admin accounts ------- 5.86s
TASK: debops.bootstrap : Install custom Ansible local facts ------------- 5.08s
TASK: Gathering Facts --------------------------------------------------- 4.62s
TASK: debops.bootstrap : Gather host facts ------------------------------ 3.73s
TASK: debops.bootstrap : Create admin accounts -------------------------- 3.27s
TASK: debops.bootstrap : Set admin home directory group and permissions --- 2.63s
```


$debops

```
PLAY RECAP **********************************************************************************************************************************************************************************************************************************
other.domain.com   : ok=189  changed=0    unreachable=0    failed=0
test.domain.com     : ok=181  changed=0    unreachable=0    failed=0

TASK: debops.ferm : Configure ip(6)tables rules ----------------------- 160.19s
TASK: debops.rsyslog : Generate rsyslog configuration rules ------------ 50.85s
TASK: debops.core : Create root directories ---------------------------- 33.73s
TASK: debops.ferm : Remove ip(6)tables rules if requested -------------- 28.05s
TASK: debops.apt_preferences : Remove legacy APT preferences ----------- 21.95s
TASK: debops.core : Save local facts ----------------------------------- 19.56s
TASK: debops.tcpwrappers : Generate hosts.allow entries ---------------- 19.40s
TASK: debops.logrotate : Generate custom log rotation configuration ---- 18.87s
TASK: debops.core : Install local fact scripts ------------------------- 13.91s
TASK: debops.rsyslog : Remove custom config files when requested ------- 13.12s
```

I've attached the output from running both commands.
[debops bootstrap -u root.txt](https://github.com/carlalexander/debops-wordpress/files/1094589/debops.bootstrap.-u.root.txt)
[debops.txt](https://github.com/carlalexander/debops-wordpress/files/1094595/debops.txt)

**Comment by carlalexander** [comment]
I just tried on Digital Ocean with the latest DebOps libraries. I didn't run into any issues. What were configuration options that you used for your server?

**Comment by r4nd0m123** [comment]
Configuration options for DebOps-WordPress or the server that is hosting the WordPress site?

For DebOps-WordPress `wordpress__ssl: True` in inventory/host_vars/test.domain.com/vars.yml As for the server I've only added the needed key and performed a system update.

UPDATE: I've narrowed the issue down to the Vultr Ubuntu 16.04.2 LTS install image. When I manually install Ubuntu Server from the ISO file I don't encounter any errors. Vultr must of added or removed something DebOps doesn't like. Perhaps the wiki should suggest manually installing or creating your own Ubuntu Server base image when not using Digital Ocean?

I have since created my own base image in Vultr for DebOps-WordPress usage.

**Comment by carlalexander** [comment]
Yeah, I'm still in the early process of creating wikis for other hosts than Digital Ocean. It's just a pretty time-consuming process and I have some more pressing documentation to write. 😞

**Comment by r4nd0m123** [comment]
Would you like a hand with this? Perhaps there needs to be some kind of table that states which provider images work out of the box?

**Comment by carlalexander** [comment]
Yeah, that would be great :)

**Comment by r4nd0m123** [comment]
I'll close this issue as it's just easier and more reliable to roll your own image than depending on the host provider.

**Comment by leclaeli** [comment]
I too have run into this problem. I am using an AWS Lightsail instance of Ubuntu 16.04.2. Setting `wordpress__ssl: True` fails at the Create WordPress ACME challenge directory task. Setting `wordpress__ssl: False` works.

**Comment by carlalexander** [comment]
Ok, I haven't tested this project on AWS much. What's the difference between lightsail and ec2?

**Comment by leclaeli** [comment]
Lightsail is a GUI friendly/easy way to spin up a VPS with an OS and/or applications already installed. https://amazonlightsail.com/.

UPDATE: I just manually added the pki-acme user and group as mentioned in [#133](https://github.com/carlalexander/debops-wordpress/issues/133#issuecomment-277800884). I then re-ran `debops wordpress` without errors.

My site now directs to https, but I get an insecure connection notice in the browser so I'm guessing something isn't right with the cert?

I also tried running everything on an ec2 instance and pretty much got the same results.
